doc/howto_release.md

```diff
@@ -229,14 +237,17 @@ Save the modified draft, but do not publish the release yet.
 
 Use the dedicated `update_version.py` script to edit the VERSION file.
 
+:loudspeaker: TODO: the lines below are sill a bit confusing, better to split
+              into subsections.
```

Comment:
This took me some time to figure out. :sweat_smile: The original text says from 9caec5ce1cb5bb5d6fac312125af41e605db8044:

> After a (final) release, switch to development version for the next micro, minor, or major version, e.g., for micro version, use:
> ...

Here is a procedure for from 8.4.0dev through RC1, RC2 and 8.4.0 to 8.4.1dev:

```bash
$ ./utils/update_version.py rc 1
read:
  user_message: Use the provided messages for the commit and the tag.
  note: Once all checks pass, you are expected to create a tag.
use:
  commit_message: 'version: GRASS GIS 8.4.0RC1'
  tag_message: 'GRASS GIS 8.4.0RC1'
$ ./utils/update_version.py dev
read:
  user_message: Use the provided message for the commit
use:
  commit_message: 'version: Back to 8.4.0dev'
$ ./utils/update_version.py rc 2
read:
  user_message: Use the provided messages for the commit and the tag.
  note: Once all checks pass, you are expected to create a tag.
use:
  commit_message: 'version: GRASS GIS 8.4.0RC2'
  tag_message: 'GRASS GIS 8.4.0RC2'
$ ./utils/update_version.py dev
read:
  user_message: Use the provided message for the commit
use:
  commit_message: 'version: Back to 8.4.0dev'
$ ./utils/update_version.py release
read:
  user_message: Use the provided messages for the commit and the tag.
  note: Once all checks pass, you are expected to create a tag.
use:
  commit_message: 'version: GRASS GIS 8.4.0'
  tag_message: 'GRASS GIS 8.4.0'
$ ./utils/update_version.py micro
read:
  user_message: Use the provided message for the commit
use:
  commit_message: 'version: Start 8.4.1dev'
```

After every RC, `./utils/update_version.py dev` - this is and should be here. Update to the next RC or final - that's happening at the beginning of the procedure, not here.

Where the confusion happens is the update between version numbers and that's where a different step is needed (not an extra step). I missed that note and created another section at the end 26d1b2ebe06bb3b857e43a4db9bba177becb5eab likely because I just kept the original heading from before the `./utils/update_version.py` script which is "Reset include/VERSION file to git development version" not "Update VERSION file".

"Update VERSION file" is I think the heading we need now. With three subsections: "After RC", "After final release" and "Commit the updated file".

The section "Update VERSION file to next version number" at the end of this file can disappear (it is the second command here).

Does that make sense to you?